catalog/transformer/catalog-transformer-csv-queryresponse/src/main/java/ddf/catalog/transformer/csv/CsvMetacardTransformer.java

```diff
@@ -64,4 +63,15 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
         writeMetacardsToCsv(Collections.singletonList(metacard), descriptors, aliases);
     return createResponse(appendable);
   }
+
+  private List<String> getColumnOrder(Map<String, Serializable> arguments) {
+    if (arguments.get("columnOrder") instanceof String) {
+      String attributeString =
+          arguments.get("columnOrder") != null ? (String) arguments.get("columnOrder") : "";
```

Comment:
❓ Do we really need a null check here? The `instanceof` predicate would fail a null. You also don't need an `else` as the first block returns.

Just a suggestion:
```
    if (arguments.get("columnOrder") instanceof String) {
      String[] attributes =
          Optional.of(arguments.get("columnOrder")).map(String.class::cast).orElse("").split(",");
      return Arrays.asList(attributes);
    }
    return Optional.of(arguments.get("columnOrder"))
        .filter(value -> value instanceof List)
        .map(value -> (List<String>) value)
        .orElse(new ArrayList<>());
```

Reply:
The code snippet in the suggestion looks correct to me 👍